exonum-java-binding-fakes/src/main/java/com/exonum/binding/fakes/mocks/MockInteraction.java

```diff
@@ -0,0 +1,91 @@
+package com.exonum.binding.fakes.mocks;
+
+import com.google.gson.Gson;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Helper class to make testing callbacks from native code easier. It allows simple tracking of
+ * interaction with a specific method of a mocked object and then return it (by calling
+ * getInteractions() method from native side) as a JSON string.
+ */
+public class MockInteraction {
+  private final String[] arguments;
+  private final List<String> interactions;
+
+  private MockInteraction(String[] arguments) {
+    this.arguments = arguments;
+    interactions = new ArrayList<>();
+  }
+
+  /**
+   * Creates new instange of {@link MockInteraction} configured with expected names of arguments for
+   * a mocked method.
+   *
+   * @param expectedArguments Names of expected arguments for a mocked method in their right order.
+   * @return New and configured instance of {@link MockInteraction}
+   */
+  public static MockInteraction createInteraction(String[] expectedArguments) {
+    return new MockInteraction(expectedArguments);
+  }
+
+  /**
+   * Custom Answer that is used to track interactions with a particular method of mocked object.
+   */
+  private class InteractionAnswer implements Answer {
```

Comment:
Please add tests for this class.

Reply:
Doesn't it look like overhead to test our internal class created for testing our code? 🙃

Reply:
It's up to you, but StringBuilder construction logic seems error-prone. However we'd get an error in tests where this class is used, so maybe it's indeed excessive. I'd wait for someone else's opinion.

Reply:
I don't think tests are necessary here